### DISCUSSION TITLE
reporting issues...

### DISCUSSION THREAD

**Discussion by babakfp** [lumeland/lume#576] [General]
Hi

[This blog post](https://lume.land/blog/posts/lume-cms/) says this is how to import lume cms:

```ts
import lumeCMS from "lume_cms/mod.ts";
```

and [this docs page](https://lume.land/cms/) says this:

```ts
import lumeCMS from "lume/cms.ts";
```

Which one is it?!!

---

I'm also getting this error in the console:

```
NotFound: The system cannot find the file specified. (os error 2): open './_site/service-worker.js'
```

---

When choosing all plugins, I get this error (the whole thing breaks and nothing works):

```
deno task lume
Task lume echo "import 'lume/cli.ts'" | deno run -A -
Loading config file file:///C:/Users/babak/OneDrive/Desktop/_/2/lumeCMS/_config.ts
Error: Missing configuration for Netlify CMS: decap_cms
    at https://deno.land/x/lume@v2.1.0/plugins/decap_cms.ts:63:15
    at Events.dispatchEvent (https://deno.land/x/lume@v2.1.0/core/events.ts:46:19)   
    at Site.#buildPages (https://deno.land/x/lume@v2.1.0/core/site.ts:637:25)        
    at eventLoopTick (ext:core/01_core.js:65:7)
    at async Site.build (https://deno.land/x/lume@v2.1.0/core/site.ts:511:9)
    at async build (https://deno.land/x/lume@v2.1.0/cli/build.ts:21:3)
    at async Command.fn (https://deno.land/x/lume@v2.1.0/cli.ts:147:5)
    at async Command.execute (https://deno.land/x/cliffy@v0.25.7/command/command.ts:1794:7)
    at async Command.parseCommand (https://deno.land/x/cliffy@v0.25.7/command/command.ts:1639:14)
    at async https://deno.land/x/lume@v2.1.0/cli.ts:156:3
```

---

When not choosing any plugins (but choosing the CMS) I get this error (it doesn't crash, things still work):

```
NotFound: The system cannot find the file specified. (os error 2): open './_site/service-worker.js'
```

**Comment by oscarotero** [comment]
Sorry, the correct specifier is

```js
import lumeCMS from "lume/cms.ts";
```

```
NotFound: The system cannot find the file specified. (os error 2): open './_site/service-worker.js'
```

This is not related with Lume.

```
Error: Missing configuration for Netlify CMS
```
Seems like you are using decapCms, a different Cms. This cms requires a different configuration file. See https://lume.land/plugins/decap_cms/

**Reply by babakfp** [reply]
Also, this is what the CLI installs:

```ts
import lumeCMS from "lume/deps/cms.ts";
```

---

About this error:

```
NotFound: The system cannot find the file specified. (os error 2): open './_site/service-worker.js'
```

I can reproduce it by:

1. `deno run -Ar https://deno.land/x/lume/init.ts`

2.
```
? Choose the configuration file format » _config.ts (TypeScript)
? Do you want to install some plugins now? » Maybe later
? Do you want to setup a CMS? » Yes
```

3. `deno task lume cms`
4. In `_cms.ts`, change `import lumeCMS from "lume/deps/cms.ts"` to `import lumeCMS from "lume/cms.ts"`
5. Done. It shows that error in the console.

---

> Seems like you are using decapCms

Ah, I see. I was expecting it to work and couldn't think of that. Maybe it's not a bad idea to add a text in the CLI showing which plugins require some configuration so the users that are new to this (like me) don't have unexpected results.

**Reply by oscarotero** [reply]
```
NotFound: The system cannot find the file specified. (os error 2): open './_site/service-worker.js'
```

This error is because your browser is requesting this file and it's not found. Probably you have run a website which register a service worker in localhost, and this affects to any other site under the same domain.
In the browser devtools > Application > Service workers, you will see this worker registered and can you remove it.

---

About the wrong specifier `import lumeCMS from "lume/deps/cms.ts";`, it's a bug in Lume. It will be fixed in the next version, sorry for the trouble. Changing manually to `lume/cms.ts` should work.
